gpflow/kernels.py

```diff
@@ -122,14 +124,15 @@ def _slice(self, X, X2):
         :return: Sliced X, X2, (Nxself.input_dim).
         """
         if isinstance(self.active_dims, slice):
-            X = X[:, self.active_dims]
+            X = slice_final_dim(X, self.active_dims, self.input_dim)
             if X2 is not None:
-                X2 = X2[:, self.active_dims]
+                X2 = slice_final_dim(X2, self.active_dims, self.input_dim)
         else:
-            X = tf.transpose(tf.gather(tf.transpose(X), self.active_dims))
+            X = tf.gather(X, self.active_dims, axis=-1)
```

Comment:
`X = X[..., self.active_dims]` ?

Reply:
I've looked again and it does work for slice but not for the gather